components/spocos.js

```diff
@@ -0,0 +1,330 @@
+import styled from 'styled-components';
+import React, { useState, useEffect } from 'react';
+import Card, {
+  CardHeader,
+  CardTitle,
+  CardContent,
+  CardButtonContainer,
+  CardDiv,
+  TableDiv,
+} from './card';
+import Button from './button';
+import { EDIT, NEW, VIEW, DELETE, COLOR, SPONSORSHIP } from '../constants';
+import {
+  getSponsors,
+  deleteSponsorImagefromStorage,
+  deleteSponsor,
+  updateSponsor,
+  uploadSponsorImageToStorage,
+} from '../utility/firebase';
+import { useAuth } from '../utility/auth';
+import Modal, {
+  Label,
+  ModalContent,
+  ModalField,
+  LogoImage,
+  UploadContainer,
+} from './modal';
+
+// TODO: move font family higher, reduce need to redefine
+
+const Text = styled.p`
+  padding-right: 80px;
+  flex: 1;
+  font-size: 16px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+const Actions = styled.div`
+  padding-right: 80px;
+  flex: 1;
+  display: flex;
+  justify-content: flex-start;
+`;
+
+export default function SponsorshipPage(props) {
+  const inputFile = React.createRef();
+
+  const [sponsors, setSponsors] = useState({});
+
+  const [newobj, setnewobj] = useState({
+    name: '',
+    link: '',
+    imgURL: '',
+    imgName: '',
+    lastmod: '',
+    lastmodBy: '',
+    tier: '',
+  });
+  const [imgObject, setimgObject] = useState({});
+  const [isLoading, setLoading] = useState(true);
+  const [imgFile, setimgFile] = useState(null);
+  const [showEditWindow, setShowEditWindow] = useState(false);
+  const [modalAction, setmodalAction] = useState({});
+  const { email: user } = useAuth().user;
+
+  const loadFirebase = async () => {
+    const data = await getSponsors(props.name);
+    setSponsors(data);
+  };
+
+  useEffect(async () => {
+    await loadFirebase();
+    setLoading(false);
+  }, []);
```

Comment:
The 'destroy is not a function' error came from here most likely. You cannot call an async function directly in an useEffect hook because the hook expects a cleanup function to be returned, but async functions automatically return a promise. 

This should work:
```suggestion
  useEffect(() => {
    const initializeFirebase = async () => {
       await loadFirebase()
       setLoading(false)
    }
    
    initializeFirebase()
  }, []);
```